File: python/day1/day1-2.py
def calcfuel(line, total):
	fuel = line // 3 - 2
	
	if(fuel <= 0):
		return total

	total += fuel
	return calcfuel(fuel, total)

total = 0
f = open('input.txt', 'r')
for line in f:
	total = calcfuel(int(line), total)
print(total)

File: cpp/day1/day1-2.cpp
#include <fstream>
#include <iostream>

using namespace std;

int calculateFuel(int num, int total);

int main()
{
	ifstream fin;
	fin.open("input.txt");

	int total = 0;

	string num;
	while(getline(fin, num))
	{
		total = calculateFuel(stoi(num), total);
	}

	cout << total << endl;
	fin.close();

	return 0;
}

int calculateFuel(int num, int total)
{
	int fuel = num / 3 - 2;
	if(fuel <= 0) return total;

	total += fuel;
	return calculateFuel(fuel, total);
}